src/gui/tray/syncstatusmodel.cpp

```diff
@@ -0,0 +1,212 @@
+    }
+    overallPercent = qBound(0, overallPercent, 100);
+    return overallPercent / 100.0;
+}
+
+void SyncStatusModel::onFolderProgressInfo(const ProgressInfo &progress)
+{
+    const qint64 completedSize = progress.completedSize();
+    const qint64 currentFile = progress.currentFile();
+    const qint64 completedFile = progress.completedFiles();
+    const qint64 totalSize = qMax(completedSize, progress.totalSize());
+    const qint64 totalFileCount = qMax(currentFile, progress.totalFiles());
+
+    setSyncProgress(calculateOverallPercent(totalFileCount, completedFile, totalSize, completedSize));
+
+    if (totalSize > 0) {
+        const auto completedSizeString = Utility::octetsToString(completedSize);
+        const auto totalSizeString = Utility::octetsToString(totalSize);
+
+        if (progress.trustEta()) {
+            setSyncString(tr("%1 of %2 · %3 left")
+                                    .arg(completedSizeString, totalSizeString)
+                                    .arg(Utility::durationToDescriptiveString1(progress.totalProgress().estimatedEta)));
+        } else {
+            setSyncString(tr("%1 of %2").arg(completedSizeString, totalSizeString));
+        }
+    }
+
+    if (totalFileCount > 0) {
+        setSyncStatusString(tr("Syncing file %1 of %2").arg(currentFile).arg(totalFileCount));
+    }
+}
+
+void SyncStatusModel::setSyncing(bool value)
+{
+    if (value == _isSyncing) {
+        return;
+    }
+
+    _isSyncing = value;
+    emit syncingChanged();
+}
+
+QString SyncStatusModel::syncString() const
+{
+    return _syncString;
+}
+
+void SyncStatusModel::setSyncString(const QString &value)
+{
+    if (_syncString == value) {
+        return;
+    }
+
+    _syncString = value;
+    emit syncStringChanged();
+}
+
+void SyncStatusModel::setSyncProgress(double value)
+{
+    if (_progress == value) {
+        return;
+    }
+
+    _progress = value;
+    emit syncProgressChanged();
+}
+
+void SyncStatusModel::setSyncStatusString(const QString& value)
+{
+    if (_syncStatusString == value) {
+        return;
+    }
+
+    _syncStatusString = value;
+    emit syncStatusStringChanged();
+}
+
+QString SyncStatusModel::syncStatusString() const
+{
+    return _syncStatusString;
+}
+
+QString SyncStatusModel::syncStatusDetailString() const
+{
+    return _syncStatusDetailString;
+}
+
+void SyncStatusModel::setSyncIcon(const QUrl &value)
+{
+    if (_syncIcon == value) {
+        return;
+    }
+
+    _syncIcon = value;
+    emit syncIconChanged();
+}
+
+void SyncStatusModel::setSyncStatusDetailString(const QString &value)
+{
+    if (_syncStatusDetailString == value) {
+        return;
+    }
+
+    _syncStatusDetailString = value;
+    emit syncStatusDetailStringChanged();
+}
+
+void SyncStatusModel::connectToFoldersProgress(const Folder::Map &folderMap)
+{
+    for (const auto &folder : folderMap) {
+        connect(folder, &Folder::progressInfo, this, &SyncStatusModel::onFolderProgressInfo, Qt::UniqueConnection);
```

Comment:
we really need to do that ?
we do not get a summary signal from `Folder::Map` ?

Reply:
Could you elaborate? I'm not sure I understand what you mean.